### PR TITLE
wip: Set permissions in artifact

### PR DESCRIPTION
**Pull request by sulemandiwan**
EDIT(bhock): Resolves AEIM-1364

### CODE REVIEW

File: lib/fauxpaas/artifact_builder.rb [hunk 1]
```diff
@@ -38,5 +39,36 @@ def add_reference(ref, path)
         .write
     end
 
+		# Set the mode bits on all files in the Artifact
+		# @param path [Pathname]
+		# @param asset_dir [String]
+		def set_access_control(path, asset_dir = "public")
+			asset_path = path/asset_dir
```
Comment (botimer) [inline lib/fauxpaas/artifact_builder.rb]: Indentation problems -- use 2 spaces. Rubocop can help, but this looks like the product of bad editor settings (`set noexpandtab` and `set ts=8` in place?).

File: lib/fauxpaas/artifact_builder.rb [hunk 2]
```diff
@@ -38,5 +39,36 @@ def add_reference(ref, path)
...
+          if File.directory?(found_path)
+            FileUtils.chmod(2770, found_path)
+          else
+            FileUtils.chmod(0660, found_path)
+          end
+        end
+      end
+
+      # If asset directory exists set permissions on it also
+      if Dir.exists?(asset_path)
```
Comment (aelkiss) [inline lib/fauxpaas/artifact_builder.rb]: I'd extract this as a separate function, which also allows us to put the `Dir.exist?` check in a "guard clause", which reduces one level of nesting:

```ruby

def chmod_path(base_dir, file_mode, dir_mode)
  return unless Dir.exist?(base_dir)

  Find.find(base_dir) do |path|
  if File.directory?(path)
    FileUtils.chmod(dir_mode, path)
  else
    FileUtils.chmod(file_mode, path)
  end
 
end
```

Comment (aelkiss) [reply lib/fauxpaas/artifact_builder.rb]: Then you could rewrite `set_access_control` as:

```ruby

def set_access_control(path)
  chmod_path(path,file_mode: 0660, dir_mode: 2770)
  chmod_path(path/"public",file_mode: 0664, dir_mode: 2775)
end
```

**Comment by malakai97** [comment]
Resolves AEIM-1364

File: lib/fauxpaas/artifact_builder.rb (hunk 2 above)
Comment (sulemandiwan) [reply lib/fauxpaas/artifact_builder.rb]: Yup. Looking into that. And thanks.

**Comment by malakai97** [comment]
Replaces by #73

**Pull request closed without merging**